contracts/gateway/OSTPrime.sol

```diff
@@ -105,64 +112,99 @@ contract OSTPrime is UtilityToken, OSTPrimeConfig {
      *      On setup of the auxiliary chain the base tokens need to be
      *      transferred in full to OSTPrime for the base tokens to be
      *      minted as OST'
+     *
+     * @return success_ `true` if initialize was successful.
      */    
     function initialize()
         public
         payable
+        returns (bool success_)
     {
-        require(msg.value == TOKENS_MAX);
+        require(
+            initialized != true,
+            "Contract is already initialized."
+        );
+
+        require(
+            msg.value == TOKENS_MAX,
+            "Payable amount must be equal to total supply of token."
+        );
+
         initialized = true;
+
+        success_ = true;
     }
 
     /**
-     * @notice convert the OST utility token to base token
+     * @notice convert the OST' EIP-20 token to OST` base token.
      *
-     * @param _amount Amount of basetoken
+     * @param _amount Amount of EIP-20 to convert to base token.
      *
-     * @return `true` if claim was successfully progressed
+     * @return success_ `true` if unwrap was successful.
      */
-    function claim(
+    function unwrap(
         uint256 _amount
     )
         public
         onlyInitialized
-        returns (bool /* success */)
+        returns (bool success_)
     {
+        require(
+            _amount > 0,
+            "Amount should not be zero."
+        );
+
         require(
             _amount <= balances[msg.sender],
-            "Insufficient balance"
+            "Insufficient balance."
         );
 
         assert(address(this).balance >= _amount);
 
-        transfer(address(this),_amount);
+
+        balances[msg.sender] = balances[msg.sender].sub(_amount);
+        balances[address(this)] = balances[address(this)].add(_amount);
```

Comment:
We can discuss this. Using transferFrom is messy.
- Currently transfer is public function, we cannot call it as msg.sender will change.
- We need to create an internal function with transfer code, so that it can be called by this class and also from the public tranferFrom function.
- TransferFrom requires the approval. This will anyhow make the code duplicate.

I tried it in my first approach but I felt that this looks better. Although it has 3 lines of duplicate code. Let me know if you disagree, I will make this change again.

Reply:
Done. Implemented `transferBalance` function.